### PR TITLE
added 4 solutions

### PR DESCRIPTION
**Pull request by olehedza**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,58 @@
...
+        }
+
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int matrixShape = matrix.length;
+        int matrixSize = matrix.length * matrix.length;
+        int min = Integer.MAX_VALUE;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve without Integer.MAX_VALUE;

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,58 @@
...
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int matrixShape = matrix.length;
+        int matrixSize = matrix.length * matrix.length;
+        int max = Integer.MIN_VALUE;
+        int rowIndex;
+        int colIndex;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need all these variables?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,58 @@
...
+
+        for (int i = 0; i < matrixSize; i++) {
+            rowIndex = i / matrixShape;
+            colIndex = i % matrixShape;
+            if (matrix[rowIndex][colIndex] < min) {
+                min = matrix[rowIndex][colIndex];
+            }
+        }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: remove empty line

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/MatrixOperation.java]: All redundant variables are terminated

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done!

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,18 +6,46 @@
...
+        for (int i = 0, j = diagonalLength - 1; i < diagonalLength; i++, j--) {
+            counterDiagonal[i] = matrix[i][j];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int matrixShape = matrix.length;
+        int matrixSize = matrix.length * matrix.length;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: Do we need these variables?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -6,18 +6,46 @@
...
+        int min = matrix[0][0];
+
+        for (int i = 1; i < matrixSize; i++) {
+            if (matrix[i / matrixShape][i % matrixShape] < min) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: It's good that you are trying to optimize your code using only one loop , but you making sacrifice out of readability of your code. Save it for codewars and leetcode tasks) but here let's use two loops

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Redundant variables are terminated

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 5 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Got it. Two loops solution was successfully added

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 6]
```diff
@@ -6,18 +6,46 @@
...
+        for (int i = 0, j = diagonalLength - 1; i < diagonalLength; i++, j--) {
+            counterDiagonal[i] = matrix[i][j];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java:28]:
```suggestion
```

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 7]
```diff
@@ -6,18 +6,46 @@
...
+                }
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = matrix[0][0];
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java:41]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**